Add Motorcycle Left & Right, and allow returns in Label

poorlymac opened the pull request:
Just some ideas you may want to merge or implement better

- Addition of a Motorcycle icon
- A left and right icon so you can choose the correct one for the general direction
- Change the track Label to a text area so you can put returns in it and make it a bit more condensed

Picture of what it looks like
![Screen Shot 2020-10-31 at 11 19 25](https://user-images.githubusercontent.com/16620846/97766553-fb357500-1b6a-11eb-9e1c-ccf85b3fc460.png)

n.b. Allowing the use of a custom icon per track may be a better solution now I think about it

McPringle commented:
Hi @poorlymac 

Thank you very much for your PR! I have just some little things to mention:

- Can you please add the credits for the icon to the credits section in the readme file?
- Please add just one motorcycle. I think it is a more general approach to add a new feature to GPX Animator, which turns the icon automatically in the correct direction. That will work for all icons and directions. I'll add a task for this.
- For your next PR, I want to suggest you do one commit per feature. That's easier to review. Of course, this time there is not a lot of changed code, so it is okay.

Again, thank you very much, I highly appreciate it!

McPringle

McPringle commented on src/main/java/app/gpx_animator/TrackIcon.java:
```diff
@@ -28,8 +28,8 @@
     @NonNls
-    private static final String[] KEYS = {"airplane", "bicycle", "bus", "car", "jogging", "riding", "sailing", "ship", "train", "tramway",
-            "trekking"};
+    private static final String[] KEYS = {"airplane", "bicycle", "bus", "car", "jogging", "motorcycleleft", "motorcycleright",
+           "riding", "sailing", "ship", "train", "tramway", "trekking"};
```
Please add only one generic motorcycle. I created issue #228 as a generic approach to turn the icons automatically in the driving direction.

McPringle commented on src/main/resources/i18n/Messages.properties:
```diff
@@ -61,6 +61,8 @@ renderer.progress.unusedframes=Skipping unused Frame: %d/%d
 renderer.progress.zoom=computed zoom is %s
 trackicon.icon.airplane=Airplane
 trackicon.icon.bicycle=Bicycle
+trackicon.icon.motorcycleleft=Motorcycle Left
+trackicon.icon.motorcycleright=Motorcycle Right
```
Same here. just one motorcycle, please.

McPringle reviewed (changes requested):
Please add the icon credits to the readme file.

poorlymac commented:
I added credits for the icon
I added a horizontal flip checkbox that flips the image for the track as I think that is sufficient for a lot of use cases

Turning the icon during the track run is a great idea especially if we can make the transition smooth

McPringle reviewed (approved):
Thank you very much, @poorlymac, I appreciate your contribution a lot! One wish from my side: Next time please do two PRs, one PR per feature. That makes the review easier if there are no mixed features. But that is no show-stopper here because these two features are luckily not very complicated. :-)

McPringle commented:
There are two merge conflicts preventing me from automatically merging your PR. About a week ago I changed the code to use the enhanced switch format with commit 230eff9d67f842854f74ec43239669b71b0155b9. There were a lot of other commits, too, which should not touch your PR. It would be great if you can merge the `master` branch into your repo, so I can merge your PR.

McPringle reviewed (changes requested):
This PR needs to be merged with the updated master first.

poorlymac commented:
Thanks, should be done now. Thanks for your patience, still learning here

McPringle commented on src/main/java/app/gpx_animator/Renderer.java:
```diff
@@ -722,14 +723,17 @@ private Point2D drawMarker(final BufferedImage bi, final int frame) {
+                        } else {
                             drawSimpleCircleOnGraphics2D(point, g2);
                         }
+                    } catch (final IOException e) {
+                        drawSimpleCircleOnGraphics2D(point, g2);
```
A fallback in case of an exception is good, but it could result in unexpected behavior and the user did not know, why. The user selects an icon but gets just a circle without a notification. I would suggest throwing an `UserException` with a descriptive message (with a translation, so use the `resourceBundle` like in other messages). The message of an `UserException` is shown to the user, so he can react and check his settings.

poorlymac replied:
Agreed, refactored on my master

McPringle commented on src/main/java/app/gpx_animator/ui/TrackSettingsPanel.java:
```diff
@@ -355,6 +413,7 @@ public void setConfiguration(final TrackConfiguration c) {
         trimGpxStartSpinner.setValue(c.getTrimGpxStart());
         trimGpxEndSpinner.setValue(c.getTrimGpxEnd());
         trackIconComboBox.setSelectedItem(c.getTrackIcon());
+        flipIcon.setSelected(c.getFlipIcon());
```
I think the value of `inputIconFileSelector` should be set here, like a few lines above with the `inputGpxFileSelector` as an example. Otherwise loading a saved configuration will lose the custom icon.

McPringle commented on src/main/resources/i18n/Messages_de.properties:
```diff
@@ -51,7 +51,9 @@ option.help.tail-duration=Dauer der farblichen Hervorhebung der Streckenlinie in
 option.help.total-time=Gesamtzeit des Videos in Millisekunden; komplementär zur Beschleunigung der Echtzeit
-option.help.track-icon=blendet an der aktuellen Position statt der Standard-Markierung ein Symbol ein (gültige Werte: Airplane, Bicycle, Bus, Car, Jogging, Riding, Sailing, Ship, Train, Tramway, Trekking)
+option.help.track-icon=blendet an der aktuellen Position statt der Standard-Markierung ein Symbol ein (gültige Werte: Airplane, Bicycle, Bus, Car, Jogging, Mororrad, Riding, Sailing, Ship, Train, Tramway, Trekking)
+option.help.track-icon-file=Mit dieser Option können Sie eine Symboldatei für eine Spur festlegen
```
Just one changed word: `Mit dieser Option können Sie eine Symboldatei für eine Route festlegen`

McPringle commented on src/main/resources/i18n/Messages_de.properties:
```diff
@@ -51,7 +51,9 @@ option.help.tail-duration=Dauer der farblichen Hervorhebung der Streckenlinie in
-option.help.track-icon=blendet an der aktuellen Position statt der Standard-Markierung ein Symbol ein (gültige Werte: Airplane, Bicycle, Bus, Car, Jogging, Riding, Sailing, Ship, Train, Tramway, Trekking)
+option.help.track-icon=blendet an der aktuellen Position statt der Standard-Markierung ein Symbol ein (gültige Werte: Airplane, Bicycle, Bus, Car, Jogging, Mororrad, Riding, Sailing, Ship, Train, Tramway, Trekking)
+option.help.track-icon-file=Mit dieser Option können Sie eine Symboldatei für eine Spur festlegen
+option.help.flip-icon=Dreht das Symbol horizontal, wenn Sie eine allgemeine Richtung anzeigen möchten
```
Just changed wording: `Spiegelt das Symbol horizontal, um eine andere allgemeine Richtung anzuzeigen`

McPringle commented on src/main/resources/i18n/Messages_de.properties:
```diff
@@ -213,10 +215,13 @@ ui.panel.generalsettings.zoom.label=Vergrösserung
 ui.panel.tracksettings.button.remove=Route entfernen
 ui.panel.tracksettings.color.label=Farbe
 ui.panel.tracksettings.gpxfile.format.all=Alle unterstützten Routingdaten
-ui.panel.tracksettings.gpxfile.format.gpx=GPX Dateien
-ui.panel.tracksettings.gpxfile.format.gpxgz=Komprimierte GPX Dateien
+ui.panel.tracksettings.gpxfile.format.png=GPX Dateien
```
I guess this change here is unwanted. PNG is not a GPX file and the removed two translations are needed for the GPX file selector dialog.

McPringle commented on src/main/resources/i18n/Messages_de.properties:
```diff
@@ -213,10 +215,13 @@ ui.panel.generalsettings.zoom.label=Vergrösserung
+ui.panel.tracksettings.gpxfile.format.png=GPX Dateien
 ui.panel.tracksettings.gpxfile.label=Routingdatei
 ui.panel.tracksettings.icon.label=Symbol
+ui.panel.tracksettings.iconfile.format.all=Alle unterstützten PNG-Dateien
```
Changed wording: `Alle unterstützten Grafikdateien`

McPringle commented on src/main/resources/i18n/Messages_de.properties:
```diff
@@ -213,10 +215,13 @@ ui.panel.generalsettings.zoom.label=Vergrösserung
+ui.panel.tracksettings.iconfile.format.all=Alle unterstützten PNG-Dateien
+ui.panel.tracksettings.iconfile.format.png=PNG-Dateien
+ui.panel.tracksettings.iconfile.label=oder wählen Sie Symboldatei
+ui.panel.tracksettings.flip.label=Flip
```
Changed wording: `Symbol spiegeln`

poorlymac replied:
Google translate :-P, fixed!

McPringle reviewed (changes requested):
Good morning! Thank you very much for your update, very good. Just two very small issues left: One with the catch block of an exception, to inform the user and the other with changed wording in the translation file. All the other stuff looks very nice!
